### PR TITLE
Implementation of setLibraryList API

### PR DESCRIPTION
**Pull request by worksofliam**
As this is this the most requested topic I get from clients, I thought it might be worth adding this API.

I have created and the API, tested it, added to the readme and added the documentation.

Let me know if there is anything else I should do.

Thanks!

Signed-off-by: Liam Allan <WorksOfLiam@users.noreply.github.com>

### CODE REVIEW

**Comment by kadler** [comment]
To this day it boggles my mind how this seemingly simple task escapes so many people.

Regardless, it seems to be something we should add if it will make things easier and discoverable for people.

File: README.md [hunk 1]
```diff
@@ -181,6 +182,24 @@ runSqlExample().catch((error) => {
...
 
+### setLibraryList
+
+Change to system naming and set the library list (using `CHGLIBL`) of the connection.
+
+```javascript
+async function execExample() {
+  const connection = new Connection({ url: '*LOCAL' });
+
+  await connection.setLibraryList(['QIWS']);
```
Comment (kadler) [inline README.md]: Might be better to show an example with an array of multiple libraries just to be explicit here

File: docs/README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
   - [**Connection.isConnected()**](#connectionisconnected)
   - [**Connection.setConnAttr(attribute, value)**](#connectionsetconnattrattribute-value)
   - [**Connection.validStmt(sql)**](#connectionvalidstmtsql)
+  - [**Connection.setLibraryList(sql)**](#connectionsetlibrarylistlist)
```
Comment (kadler) [inline docs/README.md]:
```suggestion
  - [**Connection.setLibraryList(list)**](#connectionsetlibrarylistlist)
```

File: lib/connection.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-const { dbconn } = require('idb-connector');
+const { dbconn, SQL_ATTR_DBC_SYS_NAMING, SQL_TRUE } = require('idb-connector');
+const Statement = require('./statement');
```
Comment (abmusse) [inline lib/connection.js]: I understand why you moved this up but the `const Statement = require('./statement');` was intentionally placed within getStatement() to guard against circular dependency issue.

Currently there is a way to Create a Statement from a Connection.

Connection -> Statement

There is also an unorthodox way of creating a Connection implicitly when one is not provided to the [Statement constructor](https://github.com/IBM/nodejs-idb-pconnector/blob/master/lib/statement.js#L11).

Statement -> Connection

This causes a [circular dependency issue](https://nodejs.org/api/modules.html#modules_cycles).

Running the tests after cloning your branch shows that a bunch of the Statement tests fail.

```
  4) Statement Class Tests
       bindParams
         associate parameter markers in an SQL to app variables:
     TypeError: Connection is not a constructor
      at new Statement (lib/statement.js:11:28)
      at Context.it (test/statementTest.js:90:25)


```

Comment (kadler) [reply lib/connection.js]: Agreed. If we want to improve how this circular dependency is solved (if that's possible), then we should do it in a separate branch. @worksofliam can you clean up this PR to have `setLibraryList` call `getStatement` instead?

Comment (abmusse) [reply lib/connection.js]: After PR https://github.com/IBM/nodejs-idb-pconnector/pull/43, statement.js now has a createConnection() which hold the require of connection.

File: docs/README.md (hunk 2 above)
Comment (abmusse) [reply docs/README.md]: Changed out sql -> list

File: README.md (hunk 1 above)
Comment (abmusse) [reply README.md]: Added `await connection.setLibraryList(['QIWS', 'QXMLSERV']);`

**Comment by abmusse** [comment]
The commits above cleaned up the PR and added test case for setLibraryList

**Pull request merged**